### DISCUSSION TITLE
Where are assets and resources supposed to be?

### DISCUSSION THREAD

**Discussion by eyyyyyyy3** [tauri-apps/tauri#7178] [Q&A] [answered]
What is the intended way of managing resources like language files or settings pictures etc. I've read some about it in the fs section of tauris documentation but it really doesn't seem to make much sense that my icons or images for example should be stored in "{FOLDERID_Pictures}". I am currently storing my language files and my settings file in wolders which are stored inside src-tauri cause this is how the tauri [documentation ](https://tauri.app/v1/guides/building/resources/) does it.

**Comment by FabianLars** [comment]
If you use the resource feature, then what you're doing is fine, as long as read-only access is enough. If you need to modify the files you should look into userdata dirs (taur's path apis that start with `app_`. Unless you only care about Windows and the perUser NSIS mode in which case modifying the files in the installdir is possible.

**Reply by eyyyyyyy3** [reply]
Are you talking about [@tauri-apps/api](https://tauri.app/v1/api/js/app) or what exactly do you mean? Also is it intended to use a MVC approach within tauri where the model would be rust the view svelte for example or rather HTML5 and the controller TS/JS?

**Reply by FabianLars** [reply, marked as answer]
> Are you talking about [@tauri-apps/api](https://tauri.app/v1/api/js/app) or what exactly do you mean?

I was talking about https://docs.rs/tauri/latest/tauri/struct.PathResolver.html#method.app_config_dir (not _just_ config_dir but the others too) and https://tauri.app/v1/api/js/path#functions (`@tauri-apps/api` and `window.__TAURI__`) depending on what _you_ use.

> Also is it intended to use a MVC approach within tauri where the model would be rust the view svelte for example or rather HTML5 and the controller TS/JS?

Honestly, that's completely up to you. We tend to say that _all_ the business logic should be in rust with the frontend being as dumb as possible. But yeah, completely up to you and nobody's gonna start a witch hunt if you do everything in javascript.
